Automattic/wp-calypso
# Fix with-theme flow for premium, WooCommerce and DotORG themes

If the theme is free, premium and WooCommerce, the user will now be pointed to the designSetup page.

If the theme is dot-org, the user will be sent to /marketplace/theme/install page.

However, this PR doesn't fix the installation issues for Marketplace themes.



Related to #75335

## Proposed Changes

* Redirect the users to designSetup for free, premium and WooCommerce themes and hide the back button.
* Redirect the users to the marketplace/install page for dot-org themes.
* From a technical PoV, the user is routed based on the `theme_type` URL parameter

## Testing Instructions



* Use the live link generated in this PR.
* Open an Incognito window and go to /themes (loggedout)
* Select a free, a premium and a WooCommerce theme and copy the links from the `Pick design` button. e.g: `/start/with-theme?ref=calypshowcase&theme=ctlg&style_variation=corn`
* To simplify things, go to your browser window where you are already logged-in and go to that URL.
* Create a new website and notice that you are now redirected to /designSetup page (with the theme preselected and the back button missing).
* For dot-org themes, you'll need to go directly to /start/with-theme/user?ref=calypshowcase&theme=astra&theme_type=dot-org and create a new site.
* After purchasing a business plan, you'll be redirected to the marketplace theme install page.
* However, if you haven't purchased a Business plan, the page will display an upgrade nudge saying that you need to purchase a Business plan in order to be able to install the theme. 

## Pre-merge Checklist
